### PR TITLE
connected sidebar with category wise services

### LINKED ISSUE #106: Sidebar should contain all the categories and clicking on it should give the services
When user clicks on Milkman button of sidebar this conponent will open which will show all the milkmans.
a

c

### PR DESCRIPTION
**Pull request by ashwaniYDV**
Fixes #106 

Surge Deployment Link: https://pr-110-sahaayak.surge.sh

## Sidebar data is coming from /categories route

ad

## Clicking on particular category will redirect to CategoryWiseServices component
as

## Clicking on more details on particular service will dericet to service dashboard component where user can view all the reviews and minor details of particular service (WIP)
af

### CODE REVIEW

File: src/redux/actions/categoryActions.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { 
+    GET_ALL_CATEGORIES_PENDING, GET_ALL_CATEGORIES_SUCCESS
+  } from './type';
+  import axios from 'axios';
+  
+  const url = "http://localhost:5000";
```
Comment (harsh-jindal) [inline src/redux/actions/categoryActions.js:6]: Change this to heroku url. Also create a new folder helper and create a file url.js in it and add the heroku url there and all the actions shall import the url from there

File: src/redux/actions/serviceActions.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { GET_CATEGORY_WISE_SERVICES_PENDING, GET_CATEGORY_WISE_SERVICES_SUCCESS } from './type';
+import axios from "axios";
+
+const url = "http://localhost:5000";
```
Comment (harsh-jindal) [inline src/redux/actions/serviceActions.js:4]: here also

File: src/redux/actions/categoryActions.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    .then(res => {
+        console.log(res.data.categories);
+      dispatch({
+        type: GET_ALL_CATEGORIES_SUCCESS,
+        payload: res.data.categories
+      })
+    })
+    .catch(err => {
+      console.log(err.response);
+      //dispatch(returnErrors(err.response.data, err.response.status, 'USER_UPDATE_FAIL'));
```
Comment (harsh-jindal) [inline src/redux/actions/categoryActions.js:22]: Aren't you getting a error message from server?

**Review by harsh-jindal — CHANGES_REQUESTED** [review]

**Comment by harsh-jindal** [comment]
@ashwaniYDV Remove the empty css file. Also create only one action services which will contain both service and category data. Resolve conflicts.
Also don't create so many commits its not a good practice. Please keep it in mind from next time.

File: src/redux/actions/categoryActions.js (hunk 1 above)
Comment (ashwaniYDV) [reply src/redux/actions/categoryActions.js:6]: @harsh-jindal There were changes in server therefore i used localhost

Comment (harsh-jindal) [reply src/redux/actions/categoryActions.js:6]: @ashwaniYDV What changes? heroku server will be at auto deploy with master branch of sahaayak_server

Comment (ashwaniYDV) [reply src/redux/actions/categoryActions.js:6]: Ok bhaiya Got it!!

File: src/redux/actions/categoryActions.js (hunk 3 above)
Comment (ashwaniYDV) [reply src/redux/actions/categoryActions.js]: No errors

Comment (ashwaniYDV) [reply src/redux/actions/categoryActions.js:22]: I didn't get one but I'll surely make the error handlers

File: src/components/CategoryWiseServices/CategoryWiseServices.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    
+    componentDidMount() {
+        const category=this.props.match.params.categoryName;
+        this.props.getCategoryWiseServices(category);
+    }
+
+    render() {
+
+        const serviceList=!this.props.service.pending ? ( this.props.service.services.map((service)=>{
+            console.log(service);
```
Comment (harsh-jindal) [inline src/components/CategoryWiseServices/CategoryWiseServices.js:21]: Remove this console.log too

**Review by harsh-jindal — APPROVED** [review]

**Comment by harsh-jindal** [comment]
@ashwaniYDV I'll merge this now. But ake sure you make a PR o fix the requested changes in the next PR.

**Pull request merged**